js/stdlib/ts/cfxers.ts

```diff
@@ -181,16 +177,13 @@ export class Contract implements IContract {
         }
         debug(`cfxers:handler`, fname, {est, est_err});
         if ( est ) {
-          if ( txn.gas === undefined ) {
-            txn.gas = est.gasLimit;
+          if ( txn.gasLimit === undefined ) {
+            txn.gasLimit = est.gasLimit;
           }
           if ( txn.storageLimit === undefined ) {
             txn.storageLimit = est.storageCollateralized;
           }
         }
-        if ( txn.storageLimit === undefined || txn.storageLimit == 0 ) {
-          txn.storageLimit = 2048;
```

Comment:
In my experience, the calculation of the storage limit estimate does not always put something in when it is needed and if it isn't needed, will get refunded, so we need this to use linear state.